src/TransformerGraph.cpp

```diff
@@ -587,9 +588,15 @@ bool TransformerGraph::setTransformation(osg::Node &transformer,const std::strin
     osg::Switch *switch_node = getFrameSwitch(target);
     osg::Node *old_node = FindNode::find(*switch_node,"link");
     assert(old_node);
-    osg::Node *link = vizkit::NodeLink::create(source,target,osg::Vec4(255,0,0,255));
-    link->setName("link");
-    switch_node->replaceChild(old_node,link);
+
+    vizkit::NodeLink* nodeLink = dynamic_cast<vizkit::NodeLink*>(old_node);
```

Comment:
Does that make any sense ? Aren't all children of a frame switch node NodeLInk ?

Reply:
No, there is also the coordinate axis and description text

Reply:
Fair enough. Let me now rephrase. Isn't FindNode::find(' .... ', 'link') __expected__ to return a NodeLink ?

Because if it is not, it means that there can be situations where we would end up with multiple NodeLink nodes under the same switch, and with the same name.

Reply:
I think the better was would be to implement a new class that inherits the osg::Switch, but has own setters/getters for link,axes, and text, this way we wouldn't need to check for node names at all. But my intention was to change as less code as possible.

Reply:
OK. Re-read the code. My problem was with

~~~
NodeLink* ... = dynamic_cast
if (!nodeLink) {
}
~~~

Which suggests that `nodeLink` might not be a NodeLink, even more so that the existing nodeLink is not used for anything.

So, since you do expect nodeLink to be a NodeLink, just test for it to be null and remove the dynamic_cast

But it's a moot point, because the nodeLink va

Reply:
I understood that,
I have no idea why the "link" as a osg::Node is added in the Constructor (Without contents in master), but I guessed, there might a a reason for that.
Next is, osg functions will never give me a NodeLink Object, I always have to cast in some way.

I could

Reply:
OK ... got it now.

Please add a comment to explain all of this

~~~
/** When the node structure is created, a plain Node is added as a 'link'
 * child of switch. Replace it by a NodeLink if it has not been done yet
 */
... nodeLink = dynamic_cast ...
~~~